File: my_app/api/subscriptionApi.py

from flask import Blueprint, request, jsonify
from my_app.middleware.shopify_session import verify_shopify_session_token
from ..services import subscription_service

subscription_bp = Blueprint('subscription', __name__)

# Start trial
@subscription_bp.route('/subscription/start-trial/<int:user_id>', methods=['POST'])
@verify_shopify_session_token
def api_start_trial(user_id):
	data = request.get_json(silent=True) or {}
	plan = data.get('plan', 'standard')
	trial_days = data.get('trial_days', 15)
	notify_user = data.get('notify_user', True)
	result = subscription_service.start_trial(user_id, plan, trial_days, notify_user)
	return jsonify({'success': result})

# Convert trial to paid
@subscription_bp.route('/subscription/convert/<int:subscription_id>', methods=['POST'])
@verify_shopify_session_token
def api_convert_trial(subscription_id):
	data = request.get_json(silent=True) or {}
	notify_user = data.get('notify_user', True)
	result = subscription_service.convert_trial_to_paid(subscription_id, notify_user)
	return jsonify({'success': result})

# Renew subscription
@subscription_bp.route('/subscription/renew/<int:subscription_id>', methods=['POST'])
@verify_shopify_session_token
def api_renew(subscription_id):
	data = request.get_json(silent=True) or {}
	renewal_days = data.get('renewal_days', 30)
	auto_renew = data.get('auto_renew', True)
	notify_user = data.get('notify_user', True)
	result = subscription_service.renew_subscription(subscription_id, renewal_days, auto_renew, notify_user)
	return jsonify({'success': result})

# Handle expiry
@subscription_bp.route('/subscription/handle-expiry/<int:subscription_id>', methods=['POST'])
@verify_shopify_session_token
def api_handle_expiry(subscription_id):
	data = request.get_json(silent=True) or {}
	notify_user = data.get('notify_user', True)
	result = subscription_service.handle_expiry(subscription_id, notify_user)
	return jsonify({'success': result})

# Check grace period
@subscription_bp.route('/subscription/in-grace/<int:subscription_id>', methods=['GET'])
@verify_shopify_session_token
def api_in_grace(subscription_id):
	notify_user = request.args.get('notify_user', True, type=bool)
	result = subscription_service.is_in_grace_period(subscription_id, notify_user)
	return jsonify({'in_grace': result})

File: frontend/daily-snapshot/src/api/subscriptionApi.ts


import { shopifyFetchJson } from './shopifyFetch';

const API_BASE = 'https://daily-snapshot-1.onrender.com/api/subscription';

// Types
export interface Subscription {
  id: number;
  plan: string;
  status: string;
  start_date?: string;
  end_date?: string;
  user_id?: number;
  next_billing?: string;
  payment_status?: string;
  history?: string[];
  user_info?: string;
  [key: string]: any;
}


// List all subscriptions
export function listSubscriptions() {
  return shopifyFetchJson(`${API_BASE}/list`);
}

// Get a single subscription by ID
export function getSubscription(id: number) {
  return shopifyFetchJson(`${API_BASE}/${id}`);
}

// Start trial
export function startTrial(userId: number, plan: string = 'standard', trialDays: number = 15, notifyUser: boolean = true) {
  return shopifyFetchJson(`${API_BASE}/start-trial/${userId}`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ plan, trial_days: trialDays, notify_user: notifyUser }),
  });
}

// Convert trial to paid
export function convertTrialToPaid(subscriptionId: number, notifyUser: boolean = true) {
  return shopifyFetchJson(`${API_BASE}/convert/${subscriptionId}`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ notify_user: notifyUser }),
  });
}

// Renew subscription
export function renewSubscription(subscriptionId: number, renewalDays: number = 30, autoRenew: boolean = true, notifyUser: boolean = true) {
  return shopifyFetchJson(`${API_BASE}/renew/${subscriptionId}`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ renewal_days: renewalDays, auto_renew: autoRenew, notify_user: notifyUser }),
  });
}

// Handle expiry
export function handleExpiry(subscriptionId: number, notifyUser: boolean = true) {
  return shopifyFetchJson(`${API_BASE}/handle-expiry/${subscriptionId}`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ notify_user: notifyUser }),
  });
}

// Check grace period
export function isInGracePeriod(subscriptionId: number, notifyUser: boolean = true) {
  const url = new URL(`${API_BASE}/in-grace/${subscriptionId}`);
  url.searchParams.set('notify_user', String(notifyUser));
  return shopifyFetchJson(url.toString());
}

// Update plan
export function updatePlan(subscriptionId: number, plan: string) {
  return shopifyFetchJson(`${API_BASE}/plan/${subscriptionId}`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ plan }),
  });
}

// Retry payment
export function retryPayment(subscriptionId: number) {
  return shopifyFetchJson(`${API_BASE}/retry-payment/${subscriptionId}`, {
    method: 'POST',
  });
}

// Cancel subscription
export function cancelSubscription(subscriptionId: number) {
  return shopifyFetchJson(`${API_BASE}/cancel/${subscriptionId}`, {
    method: 'POST',
  });
}

